### ISSUE TITLE
No menu set for the provided location

### ISSUE THREAD

**Issue by MKlblangenois** [wp-graphql/wp-graphql#3180]
### Description

When I try to get `menu` node using it's location, I've an error in WPGraphQL `No menu set for the provided location`, and menu return `null`
![image](https://github.com/user-attachments/assets/5443ca7e-75e3-49e0-b3f3-12fd46d60ea4)

### Steps to reproduce

1. Register a menu position
2. Create a new menu and assign it to the position
3. Add item in the menu
4. Go to GraphiQL and make the query to get the menu by it's location
```graphql
query GetMenuByLocation($location: ID!) {
  menu(id: $location, idType: LOCATION) {
    id
    databaseId
  }
}
```

### Additional context

Here's my menu position:
![image](https://github.com/user-attachments/assets/dc47b651-f8eb-4f06-a7a7-f22781253ca4)

And my GraphQL result if I query all menus with locations:
![image](https://github.com/user-attachments/assets/abd8c9e7-061e-4ca0-a28b-e39d16e144ad)

### WPGraphQL Version

1.27.2

### WordPress Version

6.6.1

### PHP Version

8.2

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have disabled ALL plugins except for WPGraphQL.

- [X] Yes
- [ ] My issue is with compatibility with a specific WordPress plugin, and I have listed all my installed plugins (and version info) above.

**Comment by josephfusco** [comment]
Hey @MKlblangenois, how are the menus registered? Can you provide us with the code you used to register the menu location?

**Comment by MKlblangenois** [comment]
Hi @josephfusco! I've a JSON with `key => value` with `position => name` and I use it like that:

```php
<?php

// JSON Exemple:
// {
// 	"menus": {
// 		"primary": "Menu principal",
// 		"topnav": "Menu haut de page"
//              ...
// 	},
// 	...
// }

class NextPressMenu extends NextPress {
	public function __construct() {
		$this->setup();

		if (!isset($this->theme_settings) || !array_key_exists('menus', $this->theme_settings)) {
			return;
		}

		$this->register_activation_hook('after_setup_theme', $this->register_theme_menus());
	}

	public function register_theme_menus() {
		foreach ($this->theme_settings['menus'] as $slug => $label) {
			register_nav_menu($slug, __($label, $this->theme_textdomain));
		}

		return true;
	}
}
```

If I can add details, querying `menuItems` by location is working fine:
![image](https://github.com/user-attachments/assets/9ab45246-fadf-4ba5-9ddd-979f2bd70f42)

**Comment by justlevine** [comment]
> Hi @josephfusco! I've a JSON with `key => value` with `position => name` and I use it like that:

Hey @MKlblangenois , assuming `$this->theme_settings['menus']` follows the format in your docblock (i.e. lowercase), your location _slug_ is either `footer_about` or `footer-about`, but from your initial screenshot, you're trying to use the string-equivalent of the Location _enum_.

Try using the slug and let us know if it works. Ref:

![image](https://github.com/user-attachments/assets/e581b816-ecf5-477d-8ee2-c553312c4e07)

@jasonbahl @josephfusco : I didn't look at the code to see what would be involved, but for DX do we maybe want to support the enum or _slug_ as the `id` when the idType is `LOCATION`? Not sure how common this particular mistake is, but generally not reading the input descriptions is common.

**Comment by jasonbahl** [comment]
@justlevine you are correct indeed, that the "slug" of the menu is expected here. I think we could _probably_ support this use case though, where either the slug or Enum would return the correct menu.

**Comment by MKlblangenois** [comment]
Hey @justlevine! Indeed, with the slug, that's working fine! 

I took what the node `menus` returned to me in `locations` and, my bad, didn't read the schema documentation enough. Thanks for your help!

I'll take this issue open in case of the support of both slug and location would like to be added.
